File: config/filters/stellar_filter_block_number.py
#!/usr/bin/env python3
import sys
import json
import logging

def main():
    try:
        # Read input from stdin instead of command line arguments
        input_data = sys.stdin.read()
        
        # Parse input JSON
        try:
            data = json.loads(input_data)
        except json.JSONDecodeError:
            return False

        # Extract ledger_number
        ledger_number = None
        if "Stellar" in data:
            ledger = data['Stellar']['ledger'].get('sequence')
            if ledger:
                ledger_number = int(ledger)

        if ledger_number is None:
            return False

        # Return True for even ledger numbers, False for odd
        result = ledger_number % 2 == 0
        print(f"Ledger number {ledger_number} is {'even' if result else 'odd'}", flush=True)
        return result

    except Exception:
        return False

if __name__ == "__main__":
    result = main()
    # Only print the final boolean result
    print(str(result).lower(), flush=True)

File: config/filters/stellar_filter_block_number.js
try {
    // Read from stdin instead of command line arguments
    let inputJson = '';
    process.stdin.on('data', chunk => {
        inputJson += chunk;
    });
    
    process.stdin.on('end', () => {
        const data = JSON.parse(inputJson);
        
        // Extract ledger sequence number
        let ledgerNumber = null;
        if (data.Stellar) {
            ledgerNumber = data.Stellar.ledger.sequence;
        }

        if (ledgerNumber === null) {
            console.log("Ledger number is None");
            console.log('false');
            return;
        }

        const result = ledgerNumber % 2 === 0;
        console.log(`Ledger number ${ledgerNumber} is ${result ? 'even' : 'odd'}`);
        console.log(result.toString());
    });

} catch (e) {
    console.log(`Error processing input: ${e}`);
    console.log('false');
}